src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Numerics/TotalOrderIeee754ComparerTests.cs

```diff
@@ -103,30 +103,32 @@ public static IEnumerable<object[]> NFloatTestData
                 yield return new object[] { 0.0f, -0.0f, 1 };
                 yield return new object[] { -0.0f, 0.0f, -1 };
                 yield return new object[] { 0.0f, 1.0f, -1 };
-                yield return new object[] { float.PositiveInfinity, 1.0f, 1 };
-                yield return new object[] { BitConverter.UInt32BitsToSingle(0x7FC00000), 1.0f, 1 };
-                yield return new object[] { BitConverter.UInt32BitsToSingle(0x7FC00000), float.PositiveInfinity, 1 };
-                yield return new object[] { float.NaN, float.NaN, 0 };
-                if (PlatformDetection.IsRiscV64Process) // float->double cast does not preserve NaN payload and sign on RISC-V
+                yield return new object[] { NFloat.PositiveInfinity, 1.0f, 1 };
+                yield return new object[] { NFloat.NaN, NFloat.NaN, 0 };
+                if (Environment.Is64BitProcess)
                 {
-                    yield return new object[] { BitConverter.UInt32BitsToSingle(0xFFC00000), float.NegativeInfinity, 1 };
-                    yield return new object[] { BitConverter.UInt32BitsToSingle(0xFFC00000), -1.0f, 1 };
-                    yield return new object[] { BitConverter.UInt32BitsToSingle(0xFFC00000), BitConverter.UInt32BitsToSingle(0x7FC00000), 0 };
-                    yield return new object[] { BitConverter.UInt32BitsToSingle(0x7FC00000), BitConverter.UInt32BitsToSingle(0x7FC00001), 0 }; // implementation defined, not part of IEEE 754 totalOrder                    
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0x7FF80000_00000000), 1.0d, 1 };
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0x7FF80000_00000000), NFloat.PositiveInfinity, 1 };
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0xFFF80000_00000000), NFloat.NegativeInfinity, -1 };
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0xFFF80000_00000000), -1.0d, -1 };
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0xFFF80000_00000000), BitConverter.UInt64BitsToDouble(0x7FF80000_00000000), -1 };
+                    yield return new object[] { BitConverter.UInt64BitsToDouble(0x7FF80000_00000000), BitConverter.UInt64BitsToDouble(0x7FF80000_00000001), -1 }; // implementation defined, not part of IEEE 754 totalOrder                    
```

Comment:
The comment here is incorrect, IEEE 754 `totalOrder` exactly defines the ordering including of NaNs.

A given implementation may not guarantee that `NaNs` are not normalized prior to `totalOrder`, but if a `NaN` of a particular shape is encountered in `totalOrder`, it does have an exact ordering in relation to a `NaN` of another shape. The same is true of some other APIs, such as `Abs` or `CopySign`.

Reply:
Done, thank you for clarification